clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy

```diff
@@ -103,4 +148,121 @@ class OpenstackServerGroupCachingAgent extends AbstractOpenstackCachingAgent {
 
     cacheResultBuilder.build()
   }
+
+  /**
+   * Helper method for creating server group.
+   * @param providerCache
+   * @param stack
+   * @param loadbalancerIds
+   * @return
+   */
+  OpenstackServerGroup buildServerGroup(ProviderCache providerCache, Stack stack, Set<String> loadbalancerIds) {
+    Map<String, Object> launchConfig = buildLaunchConfig(stack.parameters)
+    Map<String, Object> openstackImage = buildImage(providerCache, (String) launchConfig?.image)
+
+    OpenstackServerGroup.builder()
+      .account(accountName)
+      .region(region)
+      .name(stack.name)
+      .asg(buildASG(stack))
+      .launchConfig(launchConfig)
+      .loadBalancers(loadbalancerIds)
+      .image(openstackImage)
+      .buildInfo(buildInfo((Map<String, String>) openstackImage?.properties))
+      .disabled(loadbalancerIds.isEmpty()) //TODO - Determine if we need to check to see if the stack is suspended
+      .build()
+  }
+
+  /**
+   * Creates build info from image definition.
+   * @return
+   */
+  Map<String, Object> buildInfo(Map<String, String> properties) {
+    Map<String, Object> result = [:]
+
+    if (properties) {
+      String appVersionKey = properties.get('appversion')
+
+      if (appVersionKey) {
+        AppVersion appVersion = AppVersion.parseName(appVersionKey)
+
+        result.packageName = appVersion.packageName
+        result.version = appVersion.version
+        result.commit = appVersion.commit
+
+        String buildHost = properties.get('build_host')
+        String buildInfoUrl = properties.get('build_info_url')
+
+        if (appVersion.buildJobName) {
+          Map<String, String> jenkinsMap = [name: appVersion.buildJobName, number: appVersion.buildNumber]
+          if (buildHost) {
+            jenkinsMap.put('host', buildHost)
+          }
+          result.jenkins = jenkinsMap
+        }
+
+        if (buildInfoUrl) {
+          result.buildInfoUrl = buildInfoUrl
+        }
+      }
+    }
+
+    result
+  }
+  /**
+   * Builds ASG map from stack definition.
```

Comment:
Borrowed from amazon and google implementations representing an autoscaling group.